File: truss-compiler/src/main/java/truss/compiler/symbols/AbstractSymbolVisitor.java
package truss.compiler.symbols;

public class AbstractSymbolVisitor implements SymbolVisitor {
    private boolean done;

    @Override
    public boolean isDone() {
        return done;
    }

    public void setDone(boolean done) {
        this.done = done;
    }

    public void defaultVisit(Symbol symbol) {
    }

    @Override
    public void visitConstructor(ConstructorSymbol symbol) {
        defaultVisit(symbol);
    }

    @Override
    public void visitConversionOperator(ConversionOperatorSymbol symbol) {
        defaultVisit(symbol);
    }

    @Override
    public void visitDestructor(DestructorSymbol symbol) {
        defaultVisit(symbol);
    }

    @Override
    public void visitEvent(EventSymbol symbol) {
        defaultVisit(symbol);
    }

    @Override
    public void visitEventAccessor(EventAccessorSymbol symbol) {
        defaultVisit(symbol);
    }

    @Override
    public void visitField(FieldSymbol symbol) {
        defaultVisit(symbol);
    }

    @Override
    public void visitGlobal(GlobalSymbol symbol) {
        defaultVisit(symbol);
    }

    @Override
    public void visitMemberMethod(MemberMethodSymbol symbol) {
        defaultVisit(symbol);
    }

    @Override
    public void visitNamespace(NamespaceSymbol symbol) {
        defaultVisit(symbol);
    }

    @Override
    public void visitOperator(OperatorSymbol symbol) {
        defaultVisit(symbol);
    }

    @Override
    public void visitParameter(ParameterSymbol symbol) {
        defaultVisit(symbol);
    }

    @Override
    public void visitProperty(PropertySymbol symbol) {
        defaultVisit(symbol);
    }

    @Override
    public void visitPropertyAccessor(PropertyAccessorSymbol symbol) {
        defaultVisit(symbol);
    }

    @Override
    public void visitType(TypeSymbol symbol) {
        defaultVisit(symbol);
    }

    @Override
    public void visitTypeParameter(TypeParameterSymbol symbol) {
        defaultVisit(symbol);
    }

    @Override
    public void visitInvalidType(InvalidTypeSymbol symbol) {
        defaultVisit(symbol);
    }
}

File: Truss.Compiler/Symbols/AbstractSymbolVisitor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Truss.Compiler.Symbols {
    public class AbstractSymbolVisitor : ISymbolVisitor {
        public bool Done { get; set; }

        public virtual void DefaultVisit(Symbol symbol) {
        }

        public virtual void VisitConstructor(ConstructorSymbol symbol) {
            DefaultVisit(symbol);
        }

        public virtual void VisitConversionOperator(ConversionOperatorSymbol symbol) {
            DefaultVisit(symbol);
        }

        public virtual void VisitDestructor(DestructorSymbol symbol) {
            DefaultVisit(symbol);
        }

        public virtual void VisitEvent(EventSymbol symbol) {
            DefaultVisit(symbol);
        }

        public virtual void VisitEventAccessor(EventAccessorSymbol symbol) {
            DefaultVisit(symbol);
        }

        public virtual void VisitField(FieldSymbol symbol) {
            DefaultVisit(symbol);
        }

        public virtual void VisitGlobal(GlobalSymbol symbol) {
            DefaultVisit(symbol);
        }

        public virtual void VisitMemberMethod(MemberMethodSymbol symbol) {
            DefaultVisit(symbol);
        }

        public virtual void VisitNamespace(NamespaceSymbol symbol) {
            DefaultVisit(symbol);
        }

        public virtual void VisitOperator(OperatorSymbol symbol) {
            DefaultVisit(symbol);
        }

        public virtual void VisitParameter(ParameterSymbol symbol) {
            DefaultVisit(symbol);
        }

        public virtual void VisitProperty(PropertySymbol symbol) {
            DefaultVisit(symbol);
        }

        public virtual void VisitPropertyAccessor(PropertyAccessorSymbol symbol) {
            DefaultVisit(symbol);
        }

        public virtual void VisitNamedType(NamedTypeSymbol symbol) {
            DefaultVisit(symbol);
        }

        public virtual void VisitTypeParameter(TypeParameterSymbol symbol) {
            DefaultVisit(symbol);
        }

        public virtual void VisitInvalidType(InvalidTypeSymbol symbol) {
            DefaultVisit(symbol);
        }

        public void VisitArrayType(ArrayTypeSymbol symbol) {
            DefaultVisit(symbol);
        }

        public void VisitNullableType(NullableTypeSymbol symbol) {
            DefaultVisit(symbol);
        }

        public void VisitTrackedType(TrackedTypeSymbol symbol) {
            DefaultVisit(symbol);
        }

        public void VisitNakedNullableType(NakedNullableTypeSymbol symbol) {
            DefaultVisit(symbol);
        }

        public void VisitVarType(VarTypeSymbol symbol) {
            DefaultVisit(symbol);
        }
    }
}
